server/controllers/tripController.js

```diff
@@ -50,6 +61,41 @@ class TripController {
       data: saveTrip
     });
   }
+
+  /**
+   * This method handle oneWayTrip request.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async editTrip(req, res) {
+    const tripId = req.oldTrip.id;
+    const userId = req.user.id;
+    const status = 'Pending';
+    const {
+      tripType, from, to, date, reasons, accommodationId, returnDate
+    } = req.body;
+    const cityNumber = to.length;
+    let tripUpdate = {
+      tripType,
```

Comment:
Are you changing the TripType? it should be detected from your Data not from the user Entries